src/V3Sched.cpp

```diff
@@ -0,0 +1,960 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Scheduling
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2003-2022 by Wilson Snyder. This program is free software; you
+// can redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#include "config_build.h"
+#include "verilatedos.h"
+
+#include "V3Ast.h"
+#include "V3EmitCBase.h"
+#include "V3EmitV.h"
+#include "V3Order.h"
+#include "V3Sched.h"
+#include "V3Stats.h"
+#include "V3UniqueNames.h"
+
+#include "unordered_map"
+#include "unordered_set"
+
+namespace V3Sched {
+
+LogicByScope LogicByScope::clone() const {
+    LogicByScope result;
+    for (const auto& pair : *this) {
+        std::vector<AstActive*> clones;
+        for (AstActive* const activep : pair.second) clones.push_back(activep->cloneTree(false));
+        result.emplace_back(pair.first, std::move(clones));
+    }
+    return result;
+}
+
+namespace {
+
+LogicClasses gatherLogicClasses(AstNetlist* netlistp) {
+    LogicClasses result;
+
+    const auto moveIfNotEmpty
+        = [](LogicByScope& lbs, AstScope* scopep, std::vector<AstActive*>& vec) {
+              if (!vec.empty()) lbs.emplace_back(scopep, std::move(vec));
+          };
+
+    netlistp->foreach<AstScope>([&](AstScope* scopep) {
+        std::vector<AstActive*> empty;
+        std::vector<AstActive*> statics;
+        std::vector<AstActive*> initial;
+        std::vector<AstActive*> final;
+        std::vector<AstActive*> comb;
+        std::vector<AstActive*> clocked;
```

Comment:
```suggestion
        std::vector<AstActive*> initials;
        std::vector<AstActive*> finals;
        std::vector<AstActive*> combs;
        std::vector<AstActive*> clocked;
```

"final" highlights as a keyword so bothered me, rest just to match the plural and I don't really care.

Reply:
As you might notice, I already had that problem with `statics` above :)